### ISSUE TITLE
GameCube (NGC) support

### ISSUE THREAD

**Issue by crxdelsol** [jasarien/Provenance#182]
I would love to see GameCube games come back alive on the Apple TV.
Obviously I have no clue how the Apple TV hardware will be able to run this, but it would be absolutely awesome.

**Comment by gingerbeardman** [comment]
It's based on https://github.com/OpenEmu/OpenEmu which does not support GameCube at this time.

**Comment by pmp174** [comment]
Work has already begun on a game cube emulator. Its still not working. But go to https://github.com/WilliamLCobb/GC4iOS to get updates. Otherwise this is probably unlikely at least for now.

**Comment by RedLucas** [comment]
Yes, this would be insane. For what it's worth, the gamecube emulator that works for _Mac_ is called [Dolphin](https://dolphin-emu.org), but its code is probably pretty far away from something that could b ported over. I certainly do wonder what the AppleTV could do with it.

**Comment by pmp174** [comment]
The devices all have the power to run dolphin and could do it smoothly but the draw back of dolphin is it requires 16 gb of virtual memory at startup ( it requires very little to run but for startup is very intensive ) IOS only allows you to allocate 2 gb of data at startup.

**Comment by iAmRenzo** [comment]
I would like to have this too. Maybe even Wii support.

**Comment by nsuchy** [comment]
Would an iPhone or Apple TV really have enough CPU power to emulate the gamecube? That'd be awesome if it could but it seems so unlikely.

**Comment by nsuchy** [comment]
@jasarien That's unfortunate to hear, I have the iPhone 7 so if you ever need someone to test a bug in a specific model, Gamecube or otherwise, feel to mention me :)

**Comment by sevdestruct** [comment]
@nuschy please direct your discussion to our discord and keep this focused specifically on the topic.

**Comment by YarenLithien** [comment]
It’s a bit old topic, but I re-ask, with the new Apple TVs and the new versions of everything, is there still no chance to put GameCube into provenance?

**Comment by gingerbeardman** [comment]
@YarenLithien see my first reply to this issue

**Comment by 0Shorty0** [comment]
@gingerbeardman
So you want to wait for implementation on  another project by another competitive Developer Team to „copy“ such features? 
Wow. That is a honest answer. 
No wonder why this here is stuck at, v1.4, when features are just copied, with a bit of customization.

Since well documented Wii and GameCube Libraries are online for years. 
e.g 
https://github.com/devkitPro/libogc

**Comment by jasarien** [comment]
@0Shorty0 Take your trolling elsewhere. I’ve already explained technical reasons why GC/Wii emulation is not being looked at for Provenance right now. 

As for your diatribe regarding the OpenEmu team — they’re not “competitive” developers, and we’re not “copying” their features. This is the world of Open Source. We work together to share ideas, information and solutions. While Provenance began life being based on OpenEmu, and continues to stay close to a similar architecture, it has diverged quite a bit.

You’re clearly in over your head if you can’t understand why the existing GameCube emulation implementations can’t be used in a performant manner on iOS without a jailbreak, as I’ve already explained. And your further compound your misunderstanding by linking to a project for writing Homebrew software for Wii and GameCube which has nothing to do with emulating a Wii or GameCube.

Away with you, you moonfruit.

**Comment by JoeMatt** [comment]
Hardly stuck, the entire Core's platform is rewritten in Swift. The only OpenEMU code left is the Core Audio part, which I've made some additions to as well.

If GameCube can run with JIT, then it will probably be ported. I'm currently in the process of porting Jaguar, PSP, Dreamcast and Saturn. All are in alpha/beta on the develop branch.

I'm also rewriting the way we handle C modules, so I can generalize how we work with cores that support RetroArch / Libretro.h, so we don't have to hand port massive make files for these more complex modern cores.

There's not even any well working OS X/ iOS Jaguar core I'm aware of other than in the Provenance 1.5 betas. I hand wrote a lot of optimized C code to get decent performance and resolve a bunch of crashes in various ROMs.

We've also supplied patches to Reicast , Mupen, GLiden64, that are now in their main repos. -We're hardly just copying.

If you would like to contribute instead of making pithy, ill informed comments, there's pretty good documentation on the core porting process in this ticket, 
https://github.com/Provenance-Emu/Provenance/issues/747#issuecomment-372468594

Feel free to port anything you'd like and make a Pull Request.

Provenance is open source and supported by a very small team.
https://github.com/Provenance-Emu/Provenance/graphs/contributors

The community gets what they put in, which, other than a few dedicated people, hasn't been a whole lot.

Most of the efforts I've been doing have been really boring things like converting most of the app to Swift, using Protocol Oriented design patterns, making build and testing tools etc - in a hopes to encourage other developers to submit more by making it as easy as possible to jump in.

I've been working at a startup since the Fall that just ran out of money and was acquired. I'm leaving to head up the iOS team at a Fortune 500 ranked tech company, so I've been a little busy.

I'm willing to take over Provenance development as a full time if someone wants to hire me to do that instead. Being in America I'll need health insurance and 3 weeks paid vacation. Company car would be nice too. If you can scrape together ~$250k a year I'll port anything you want.

**Comment by alexchandel** [comment]
Note OpenEmu supports GameCube as of v2.2.

Relevant links:
https://github.com/dolphin-emu
https://github.com/WilliamLCobb/GC4iOS
https://github.com/OpenEmu/dolphin

**Comment by JoeMatt** [comment]
Dolphin is already in develop branch.

**Comment by Bubba8291** [comment]
@JoeMatt How much would I have to donate for Gamecube to work on the Provenance tvOS version?

**Comment by gingerbeardman** [comment]
@Bubba8291 it's already available in the development branch, see the comment directly above yours!

**Comment by Bubba8291** [comment]
@gingerbeardman The sound doesn't play and the frames were lagging. Was I doing something wrong if it's supposed to not be lagging?

**Comment by danrha** [comment]
Are you guys making any progress on this? Dolphin's on the repo but it seems to be disabled on Provenance TV. I could dive deeper into getting it working if you need help :)

**Comment by JoeMatt** [comment]
we have two dolphin cores on ios and tvos, our custom jitless optimized one and the retroarch version
